lib/media/media_source_engine.js

```diff
@@ -1117,12 +1137,13 @@ shaka.media.MediaSourceEngine = class {
    *   partial read of the segment.
    * @param {boolean=} fromSplit
    * @param {number=} continuityTimeline an optional continuity timeline
+   * @param {boolean=} isInitData True if an init segment is being appended.
    * @return {!Promise}
    */
   async appendBuffer(
       contentType, data, reference, stream, hasClosedCaptions, seeked = false,
       adaptation = false, isChunkedData = false, fromSplit = false,
-      continuityTimeline) {
+      continuityTimeline, isInitData = false) {
```

Comment:
isInitData is not necessary. When reference is null --> init segments, so you can use it instead of isInitData